core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -740,6 +743,7 @@ private[spark] class TaskSetManager(
     }
     removeRunningTask(tid)
     info.markFinished(state)
+    successfulTaskDurations.remove(taskInfos(tid).duration)
```

Comment:
@kayousterhout, @squito  That would be a workaround, yes : at the expense of divergent data structures.